### PR TITLE
Added Resources config for initContainers

### LINKED ISSUE #477: Allow setting resources for kafka pod initContainers to able to control kafka pod QoS class
**Is your feature request related to a problem? Please describe.**

Kafka broker `Pod` created by the operator have the following initContainers:
- `cruise-control-reporter` to copy the cc jar into the kafka lib dir
- `jmx-exporter` to copy prometheus jmx_exporter libs into kafka lib dir

In our environment we'd like to control the resources (cpu/mem requests/limit) set for this initContainers to control the overall Kafka pod QoS class. as the operator is not setting any value, the `LimitRange` defaults are used and (for us) sets them to [1]:
```
  limits:
  - default:
      cpu: "1"
      memory: 512Mi
    defaultRequest:
      cpu: 100m
      memory: 256Mi
```

As [requests<limits the overall QoS class computed is `Burstable` ](https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/#create-a-pod-that-gets-assigned-a-qos-class-of-guaranteed)for the whole kafka pod.
We'd like to be able to control these to ensure the pod gets into `Guaranteed` class
  

[1] Kafka Pod annotation confirming this
```
kubernetes.io/limit-ranger: >-
      LimitRanger plugin set: cpu, memory request for init container
      ; cpu, memory limit for init container
      cruise-control-reporter; cpu, memory request for init container
      jmx-exporter; cpu, memory limit for init container jmx-exporter
```

### PR DESCRIPTION
**Pull request by azun**
| Q               | A      |
| --------------- | ------ |
| Bug fix?        | no |
| New feature?    | yes |
| API breaks?     | no |
| Deprecations?   | no |
| Related tickets | fixes #477  |
| License         | Apache 2.0 |

### What's in this PR?
Added initContaninerResourceRequirements to brokerConfig and cruisecontrolConfg

### Why?
When no resource requirements are specified, the broker pod is not Guaranteed for QoS scheduling 

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/developer-guide/blob/master/docs/coding-style/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: pkg/sdk/v1beta1/kafkacluster_types.go [hunk 1]
```diff
@@ -156,6 +156,8 @@ type BrokerConfig struct {
 	BrokerIngressMapping []string `json:"brokerIngressMapping,omitempty"`
 	// InitContainers add extra initContainers to the Kafka broker pod
 	InitContainers []corev1.Container `json:"initContainers,omitempty"`
+	// ResourceRequirements for initContainers
+	InitContainersResourceRequirements *corev1.ResourceRequirements `json:"initContainersResourceRequirements,omitempty"`
```
Comment (stoader) [inline pkg/sdk/v1beta1/kafkacluster_types.go:160]: This field is redundant. The `InitContainers []corev1.Container` field already defines resource requirements for each init container.

**Review by stoader — CHANGES_REQUESTED** [review]

File: pkg/sdk/v1beta1/kafkacluster_types.go [hunk 2]
```diff
@@ -196,6 +198,8 @@ type CruiseControlConfig struct {
 	CruiseControlAnnotations map[string]string `json:"cruiseControlAnnotations,omitempty"`
 	// InitContainers add extra initContainers to CruiseControl pod
 	InitContainers []corev1.Container `json:"initContainers,omitempty"`
+	// ResourceRequirements for initContainers
+	InitContainersResourceRequirements *corev1.ResourceRequirements `json:"initContainersResourceRequirements,omitempty"`
```
Comment (stoader) [inline pkg/sdk/v1beta1/kafkacluster_types.go:202]: This field is redundant. The InitContainers []corev1.Container field already defines resource requirements for each init container.

**Review by stoader — CHANGES_REQUESTED** [review]

File: pkg/resources/kafka/pod.go [hunk 3]
```diff
@@ -193,6 +193,7 @@ func getInitContainers(brokerConfigInitContainers []corev1.Container, kafkaClust
 				Name:      "extensions",
 				MountPath: "/opt/kafka/libs/extensions",
 			}},
+			Resources: *brokerConfig.InitContainersResourceRequirements,
```
Comment (amuraru) [inline pkg/resources/kafka/pod.go:196]: what we really miss here are the sensible defaults for builtin init containers.
I suggest we provide them statically here inline (via a helper method or so).

As @stoader also correctly pointed out - any additional init container can already provide resource requirements via the `InitContainers` struct.

File: pkg/resources/cruisecontrol/deployment.go [hunk 4]
```diff
@@ -203,6 +205,19 @@ func GeneratePodAnnotations(kafkaCluster *v1beta1.KafkaCluster, capacityConfig s
 	return util.MergeAnnotations(annotations...)
 }
 
+func getInitResourcesRequirements() corev1.ResourceRequirements {
```
Comment (stoader) [inline pkg/resources/cruisecontrol/deployment.go:208]: Let's move this function to a common place in order to not duplicate it. Also I'd rename it to something like `getDefaultInitContainerResourcesRequirements` as that name better reflects what it is for.

**Review by stoader — CHANGES_REQUESTED** [review]

**Review by stoader — APPROVED** [review]

**Review by hi-im-aren — APPROVED** [review]

**Pull request merged**